src/lib/components/ProgressBar.svelte

```diff
@@ -0,0 +1,51 @@
+<script lang="ts">
+	import type { Player } from '$lib';
+  import { tweened } from 'svelte/motion';
+  import { cubicOut } from 'svelte/easing';
+
+	export let player: Player;
+  export let color: string;
+
+  const progress = tweened(0, {
+    duration: 500,
+    easing: cubicOut,
+  })
+
+  $: progressPercent = player.score / player.scoreRequired
+  $: progress.set(progressPercent)
+
+</script>
+
+<div class="progress-container">
+	<div class="player-name">
+		{player.name}
+	</div>
+	<div class="progress-bar-outer">
+    <div class="progress-bar-inner" style="width: {$progress * 100}%; background-color: {color}" ></div>
+  </div>
+</div>
+
+<style>
+	.progress-container {
+		display: grid;
+		grid-template-columns: 1fr 9fr;
+		align-items: center;
+	}
+
+  .player-name {
+    margin: auto;
+  }
```

Comment:
appears unneeded. Looks the same for small screens. For larger screens, we can use margin on the `main` tag to create horizontal space on both ends